acore-docker-ac-eluna-ts-dev-1  | npm WARN deprecated tslint@6.1.3: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.

`docker compose up` can't finish in lastest version, due to the failed in ` acore-docker-ac-eluna-ts-dev-1` container:

```
...
[+] Running 6/6
 ⠿ Container acore-docker-ac-client-data-1   Created                                                               2.6s
 ⠿ Container acore-docker-ac-database-1      Created                                                               1.4s
 ⠿ Container acore-docker-ac-eluna-ts-dev-1  Created                                                               1.4s
 ⠿ Container acore-docker-ac-db-import-1     Created                                                               0.4s
 ⠿ Container acore-docker-ac-authserver-1    Created                                                               0.2s
 ⠿ Container acore-docker-ac-worldserver-1   Created                                                               0.4s
Attaching to acore-docker-ac-authserver-1, acore-docker-ac-client-data-1, acore-docker-ac-database-1, acore-docker-ac-db-import-1, acore-docker-ac-eluna-ts-dev-1, acore-docker-ac-worldserver-1
acore-docker-ac-database-1      | 2023-01-23 05:04:01+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.32-1.el8 started.
acore-docker-ac-client-data-1 exited with code 0
acore-docker-ac-database-1      | 2023-01-23 05:04:04+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
acore-docker-ac-database-1      | 2023-01-23 05:04:04+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.32-1.el8 started.
acore-docker-ac-database-1      | '/var/lib/mysql/mysql.sock' -> '/var/run/mysqld/mysqld.sock'
acore-docker-ac-database-1      | 2023-01-23T05:04:06.394956Z 0 [Warning] [MY-011068] [Server] The syntax '--skip-host-cache' is deprecated and will be removed in a future release. Please use SET GLOBAL host_cache_size=0 instead.
acore-docker-ac-database-1      | 2023-01-23T05:04:06.484539Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.32) starting as process 1
acore-docker-ac-database-1      | 2023-01-23T05:04:06.793011Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
acore-docker-ac-database-1      | 2023-01-23T05:04:10.497660Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
acore-docker-ac-database-1      | 2023-01-23T05:04:11.697895Z 0 [System] [MY-010229] [Server] Starting XA crash recovery...
acore-docker-ac-database-1      | 2023-01-23T05:04:11.781448Z 0 [System] [MY-010232] [Server] XA crash recovery finished.
acore-docker-ac-database-1      | 2023-01-23T05:04:13.188234Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
acore-docker-ac-database-1      | 2023-01-23T05:04:13.188276Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
acore-docker-ac-database-1      | 2023-01-23T05:04:13.278647Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
acore-docker-ac-database-1      | 2023-01-23T05:04:13.899271Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060, socket: /var/run/mysqld/mysqlx.sock
acore-docker-ac-database-1      | 2023-01-23T05:04:13.976242Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.32'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server - GPL.
acore-docker-ac-eluna-ts-dev-1  | npm WARN deprecated tslint@6.1.3: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.
acore-docker-ac-db-import-1     | Wrong Loggers configuration. Review your Logger config section.
acore-docker-ac-db-import-1     | Creating default loggers [root (Error), server (Info)] to console
acore-docker-ac-db-import-1 exited with code 1
service "ac-db-import" didn't completed successfully: exit 1
```

After docker compose command finished:

```
# docker ps
CONTAINER ID   IMAGE                   COMMAND                  CREATED         STATUS                            PORTS                                                    NAMES
164836aefde5   acore/eluna-ts:master   "docker-entrypoint.s…"   7 minutes ago   Restarting (243) 37 seconds ago
                                                   acore-docker-ac-eluna-ts-dev-1
cce92dd72eb3   mysql:8.0               "docker-entrypoint.s…"   7 minutes ago   Up 7 minutes (healthy)            33060/tcp, 0.0.0.0:63306->3306/tcp, :::63306->3306/tcp   acore-docker-ac-database-1
```

The container `acore-docker-ac-eluna-ts-dev-1` is always restarting, logs:

```
# docker logs acore-docker-ac-eluna-ts-dev-1
npm WARN deprecated tslint@6.1.3: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.
npm WARN checkPermissions Missing write access to /eluna-ts/modules
npm WARN notsup Unsupported engine for @azerothcore/eluna-ts-definitions@1.2.12: wanted: {"node":">=14.0.0","npm":">=7.0.0"} (current: {"node":"14.20.0","npm":"6.14.17"})
npm WARN notsup Not compatible with your version of node/npm: @azerothcore/eluna-ts-definitions@1.2.12
npm WARN notsup Unsupported engine for @azerothcore/eluna-ts-lib@1.0.13: wanted: {"node":">=14.0.0","npm":">=7.0.0"} (current: {"node":"14.20.0","npm":"6.14.17"})
npm WARN notsup Not compatible with your version of node/npm: @azerothcore/eluna-ts-lib@1.0.13
npm WARN acore-docker-ts@1.0.0 No repository field.
npm WARN acore-docker-ts@1.0.0 No license field.

npm ERR! code EACCES
npm ERR! syscall access
npm ERR! path /eluna-ts/modules
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, access '/eluna-ts/modules'
npm ERR!  [Error: EACCES: permission denied, access '/eluna-ts/modules'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/eluna-ts/modules'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2023-01-23T05_14_38_490Z-debug.log
```

--- a/.github/workflows/test-install.yml
+++ b/.github/workflows/test-install.yml
@@ -37,7 +37,7 @@ jobs:
         shell: bash
         run: |
           cp .github/docker-compose.dry-run.yml docker-compose.override.yml
-          docker compose up --attach ac-worldserver --attach ac-authserver
+          docker compose up --attach ac-db-import --attach ac-worldserver --attach ac-authserver
 
   test-dev-install-process:
     env:

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -83,7 +83,7 @@ services:
   ac-eluna-ts-dev:
     image: acore/eluna-ts:master
     # the build:lib is done in the healthcheck
-    command: sh -c "touch /eluna-ts/modules/index.ts && cd /eluna-ts/modules && npm install && cd /eluna-ts && npm run dev"
+    command: sh -c "cd /eluna-ts/modules && touch index.ts && npm install && cd /eluna-ts && npm run dev"
     # small workaround to make the other services to wait for the eluna-ts build
     healthcheck:
       test: sh -c "cd /eluna-ts/modules/eluna-ts && npm run build:lib && npm run build"
@@ -95,11 +95,6 @@ services:
     volumes:
       # from typescript
       - ./scripts/typescript:/eluna-ts/modules
-      # explicitly declare file mount so that is always existing
-      - type: bind
-        source: ./scripts/typescript/index.ts
-        target: /eluna-ts/modules/index.ts
-        read_only: false
       # to lua
       - ./scripts/lua:/eluna-ts/dist/
 

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -83,7 +83,7 @@ services:
   ac-eluna-ts-dev:
     image: acore/eluna-ts:master
     # the build:lib is done in the healthcheck
-    command: sh -c "cd /eluna-ts/modules && npm install && cd /eluna-ts && npm run dev"
+    command: sh -c "touch /eluna-ts/modules/index.ts && cd /eluna-ts/modules && npm install && cd /eluna-ts && npm run dev"
     # small workaround to make the other services to wait for the eluna-ts build
     healthcheck:
       test: sh -c "cd /eluna-ts/modules/eluna-ts && npm run build:lib && npm run build"
@@ -95,6 +95,11 @@ services:
     volumes:
       # from typescript
       - ./scripts/typescript:/eluna-ts/modules
+      # explicitly declare file mount so that is always existing
+      - type: bind
+        source: ./scripts/typescript/index.ts
+        target: /eluna-ts/modules/index.ts
+        read_only: false
       # to lua
       - ./scripts/lua:/eluna-ts/dist/
 

